pkg/controllers/sync/controller.go

```diff
@@ -415,13 +415,21 @@ func (s *SyncController) reconcile(ctx context.Context, federatedName common.Qua
 		fedResource.RecordError("EnsureFinalizerError", errors.Wrap(err, "Failed to ensure finalizer"))
 		return worker.StatusError
 	}
-
-	return s.syncToClusters(ctx, fedResource)
+	syncClusters, selectedClusters, reconcileStatus := s.ensureSyncClusters(ctx, fedResource)
+	if reconcileStatus != worker.StatusAllOK {
+		return reconcileStatus
+	}
+	return s.syncToClusters(ctx, fedResource, syncClusters, selectedClusters)
 }
 
-// syncToClusters ensures that the state of the given object is
-// synchronized to member clusters.
-func (s *SyncController) syncToClusters(ctx context.Context, fedResource FederatedResource) worker.Result {
+func (s *SyncController) ensureSyncClusters(
```

Comment:
```suggestion
// prepareToSync performs the following preprocessing steps required to sync federated objects to selected member clusters:
//   1. Compute the list of selected member clusters from the placement field.
//   2. Compute the list of member clusters that requires an operation to be dispatched.
//   3. For newly selected clusters, update the PropagationStatus for these clusters to PendingCreate.
//
// The PendingCreate status allows us to safely skip checking of clusters during object deletion when PropagationStatus is empty. If not, it might be that the object was created but we failed to update the federated object's status previously.
func (s *SyncController) prepareToSync(
```